src/optimizer.cpp

```diff
@@ -231,22 +222,42 @@ auto Optimizer::getControlFromSequence(const unsigned int offset)
   return xt::view(control_sequence_.data, offset);
 }
 
-MotionModel Optimizer::getMotionModel() const
+void Optimizer::setMotionModel(const std::string & model)
 {
-  return motion_model_t_;
-}
+  if (model == "DiffDrive") {
+    motion_model_ = std::make_unique<DiffDriveMotionModel>();
+  } else if (model == "Omni") {
+    motion_model_ = std::make_unique<OmniMotionModel>();
+  } else if (model == "Ackermann") {
+    motion_model_ = std::make_unique<AckermannMotionModel>();
+  } else {
+    throw std::runtime_error(std::string(
+        "Model %s is not valid! Valid options are DiffDrive, Omni, or Ackermann", model.c_str()));
+  }
 
-void Optimizer::setMotionModel(const MotionModel motion_model)
-{
-  motion_model_t_ = motion_model;
-  state_.idx.setLayout(motion_model);
-  control_sequence_.idx.setLayout(motion_model);
+  state_.idx.setLayout(motion_model_->isHolonomic());
+  control_sequence_.idx.setLayout(motion_model_->isHolonomic());
 }
 
 xt::xtensor<double, 3> & Optimizer::getGeneratedTrajectories()
 {
   return generated_trajectories_;
 }
 
+void Optimizer::setMaxVelocities(const double vx, const double vy, const double vtheta)
```

Comment:
Why pass it by const ? Does we care if it will change here ?  It's almost doubles the signature.

Reply:
it signals intent that these variables should not be changed. Typically I make anything const so that someone later can't come back and abuse the API (without me noticing while scanning through PR changes) and change the underlying value in some way that would create a very subtle and hard to find bug.

This is called very infrequently, so I'm not super worried about performance. This is only called on a speed limiting callbacks which should be infrequent